File: Samples/CS/02-ApplicationMenuButton/Form1.cs
using System;
using System.Windows.Forms;
using RibbonLib;
using RibbonLib.Controls;
using RibbonLib.Controls.Events;
using RibbonLib.Interop;

namespace _02_ApplicationMenuButton
{
    public enum RibbonMarkupCommands : uint 
    {
         cmdApplicationMenu = 1000,
         cmdButtonNew = 1001,
         cmdButtonOpen = 1002,
         cmdButtonSave = 1003,
         cmdButtonExit = 1004,
    }
    
    public partial class Form1 : Form
    {
        private RibbonApplicationMenu _applicationMenu;
        private RibbonButton _buttonNew;
        private RibbonButton _buttonOpen;
        private RibbonButton _buttonSave;
        private RibbonButton _buttonExit;

        public Form1()
        {
            InitializeComponent();

            _applicationMenu = new RibbonApplicationMenu(_ribbon, (uint)RibbonMarkupCommands.cmdApplicationMenu);
            _buttonNew = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonNew);
            _buttonOpen = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonOpen);
            _buttonSave = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonSave);
            _buttonExit = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonExit);

            _applicationMenu.TooltipTitle = "Menu";
            _applicationMenu.TooltipDescription = "Application main menu";

            _buttonNew.ExecuteEvent += new EventHandler<ExecuteEventArgs>(_buttonNew_ExecuteEvent);
        }

        void _buttonNew_ExecuteEvent(object sender, ExecuteEventArgs e)
        {
            MessageBox.Show("new button pressed");
        }
    }
}

File: Samples/VB/02-ApplicationMenuButton/Form1.vb
Imports RibbonLib
Imports RibbonLib.Controls
Imports RibbonLib.Controls.Events
Imports RibbonLib.Interop

Namespace _02_ApplicationMenuButton
	Public Enum RibbonMarkupCommands As UInteger
		 cmdApplicationMenu = 1000
		 cmdButtonNew = 1001
		 cmdButtonOpen = 1002
		 cmdButtonSave = 1003
		 cmdButtonExit = 1004
	End Enum

	Partial Public Class Form1
		Inherits Form
		Private _applicationMenu As RibbonApplicationMenu
		Private _buttonNew As RibbonButton
		Private _buttonOpen As RibbonButton
		Private _buttonSave As RibbonButton
		Private _buttonExit As RibbonButton

		Public Sub New()
			InitializeComponent()

			_applicationMenu = New RibbonApplicationMenu(_ribbon, CUInt(RibbonMarkupCommands.cmdApplicationMenu))
			_buttonNew = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonNew))
			_buttonOpen = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonOpen))
			_buttonSave = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonSave))
			_buttonExit = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonExit))

			_applicationMenu.TooltipTitle = "Menu"
			_applicationMenu.TooltipDescription = "Application main menu"

			AddHandler _buttonNew.ExecuteEvent, AddressOf _buttonNew_ExecuteEvent
		End Sub

		Private Sub _buttonNew_ExecuteEvent(ByVal sender As Object, ByVal e As ExecuteEventArgs)
			MessageBox.Show("new button pressed")
		End Sub
	End Class
End Namespace
